### DISCUSSION TITLE
route based validation error

### DISCUSSION THREAD

**Discussion by nbraun-wolf** [fastapi/fastapi#4069] [Questions] [answered]
Hi, I have the requirement to customize the error status codes on pydantic schema validation error. But I have only found an option to create a global validation error handler. https://fastapi.tiangolo.com/tutorial/handling-errors/#override-request-validation-exceptions

Is there a way to do it per route without having to resort to optional route arguments and manual validation?

**Comment by ghandic** [comment]
Could you show an example? You can add what errors each of the routes could return, I'd suggest having a standard global error model though

**Reply by ghandic** [reply]
The validation error is returning a 422 unprocessable entity which is always correct. What is the reason you would want to change that?

**Reply by ghandic** [reply]
By definition of what pydantic is doing it is validating to your schema that the entity is processable through initial validation checks. 

I'm just offering advise and would handle other error codes separately from pydantic validation.

Could you give a use case for when you would want to return a different 4XX error code? Imo it should be 4XX since you define your schema it would only be a client's error for sending something that doesn't agree with your schema.

Note that pydantic can raise 500 responses if there are server errors

**Reply by ghandic** [reply]
4XX means starting with 4, eg 422, 404 etc. If you want to return a 200 response code for a failure then you'll not be following HTTP or REST standards/protocols. 

I was asking for a use case or scenario that you would want to create an open dialogue or to show a counter example.

Please refer to the code of conduct as I'm offering my expertise for free for you, also please don't leave aggressive responses and then delete/edit them. I get email notifications for this repo so I can help assist the community.

**Comment by nbraun-wolf** [comment]
It seems like, I am not getting a response based on the technical possibilities. I do not want to discuss which status code is when the best. That is a different discussion and was not intended to be the scope of this question. Hence, I request to delete this question. Thank you,

**Comment by Kludex** [comment, marked as answer]
We have a PR in discussion on Starlette about route middlewares, which will be useful for your use case. But I didn't have the time check that yet, I'll have time this week I guess.

As of now, you can mount different applications I guess, or create a middleware that overrides the status code depending on the endpoint path.
